Modules/termios.c

```diff
@@ -120,37 +120,52 @@ termios_tcgetattr_impl(PyObject *module, int fd)
         v = PyBytes_FromStringAndSize(&ch, 1);
         if (v == NULL)
             goto err;
-        PyList_SetItem(cc, i, v);
+        PyList_SET_ITEM(cc, i, v);
     }
 
     /* Convert the MIN and TIME slots to integer.  On some systems, the
        MIN and TIME slots are the same as the EOF and EOL slots.  So we
        only do this in noncanonical input mode.  */
     if ((mode.c_lflag & ICANON) == 0) {
         v = PyLong_FromLong((long)mode.c_cc[VMIN]);
-        if (v == NULL)
+        if (v == NULL) {
+            goto err;
+        }
+        if (PyList_SetItem(cc, VMIN, v) < 0) {
```

Comment:
This can be replaced with `PyList_SET_ITEM`; we know that `cc` is a list.

Reply:
But what if VMIN or VTIME are out of range? We do not know this, their definition is out of our control.

Reply:
Ah, good catch. Disregard my comment.